Updates to highlighting to not rely on default tag

PrateekKuk opened the pull request.

PrateekKuk commented:
you can ignore the commit "Updated name of default tag to 'default'" because it's overwritten by the changes by the the most recent commits

amyxzhang commented:
Same question as in the eyebrowse-server pull request - it seems like we're stripping out a lot of the tag functionality. But I would not like to do that as tags are part of what we want to have in eyebrowse. Can you leave it in? The difference with your desired change is to no longer require a tag, not to remove tag functionality entirely.

PrateekKuk commented:
added everything back in - I also cleaned up the tagging code so it doesn't rely on the default tag like the old commit did. The file basically includes all the tagging code that it originally had. I also cleaned up some old work I had done on tagging, so you will see some things being removed or commented out.
